src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/LanguageServerOperationContext.cs

```diff
@@ -0,0 +1,160 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Intellisense;
+using Microsoft.PowerFx.LanguageServerProtocol.Protocol;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Handlers
+{
+    /// <summary>
+    /// A context for a language server operation with information that might be needed when handling the operation.
+    /// It also has helper methods to simplify the operation handling.
+    /// Lifetime of this context is per request.
+    /// </summary>
+    public class LanguageServerOperationContext
+    {
+        /// <summary>
+        /// A factory to create a scope for the operation.
+        /// </summary>
+        private readonly IPowerFxScopeFactory _scopeFactory;
+
+        public LanguageServerOperationContext(IPowerFxScopeFactory scopeFactory)
+        {
+            _scopeFactory = scopeFactory;
+        }
+
+        /// <summary>
+        /// Language Server Method Identifier for which this context is created.
+        /// </summary>
+        public string LspMethod { get; init; }
+
+        /// <summary>
+        /// Language Server Logger.
+        /// </summary>
+        public ILanguageServerLogger Logger { get; init; }
+
+        /// <summary>
+        /// ID of the request. This won't be relevant for notification handlers.
+        /// </summary>
+        public string RequestId { get; init; }
+        
+        /// <summary>
+        /// Raw input of the operation. Usually this is passed from the client.
+        /// </summary>
+        public string RawOperationInput { get; init; }
+
+        /// <summary>
+        /// A builder to create the output of the operation.
+        /// No handlers are expected to return a value. They should use this builder to write the response.
+        /// A handler can write multiple responses and the builder will take care of the correct format.
+        /// </summary>
+        public LanguageServerOutputBuilder OutputBuilder { get; init; }
+
+        /// <summary>
+        /// Host Task Executor to run tasks in host environment.
+        /// </summary>
+        public IHostTaskExecutor HostTaskExecutor { get; init; }
+
+        private IPowerFxScope _scope = null;
+
+        /// <summary>
+        /// Gets or creates the scope for the operation.
+        /// </summary>
+        /// <param name="uri">Uri to use for scope creation.</param>
+        /// <returns>Scope.</returns>
+        private IPowerFxScope GetScope(string uri)
+        {
+            return _scope ??= _scopeFactory.GetOrCreateInstance(uri);
+        }
+
+        /// <summary>
+        /// A helper method to convert the expression to display format.
+        /// </summary>
+        /// <param name="uri">Uri to use for check result creation.</param>
+        /// <param name="expression">Expression to create check result for.</param>
+        /// <returns>Converted expression.</returns>
+        public string ConvertToDisplay(string uri, string expression)

```

Comment:
>ConvertToDisplay [](http://example.com/codeflow?start=22&length=16)

This should probably be under lock

Reply:
No this is the method i forgot to remove